src/Chapter3.hs

```diff
@@ -374,6 +400,34 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+{-
+Actually I don't see the point in writing separate
+Monster and Knight types as their properties are the same
+-}
+data Fighter = Fighter
+    { health :: Int
+    , attack :: Int
+    , gold   :: Int
+    } deriving (Show)
+
+{- Returns the updated knight, monster and fight result (-1 knight lost, 1 knight won, 0 draw) -}
+fight :: (Fighter, Fighter)-> (Fighter, Fighter, Int)
```

Comment:
Nice function!
The only issue with the reusage of the same data type for Monster and Knight in here is that you can easily mix them when calling the function:

```haskell
> fight (knight, monster)
> fight (monster, knight)
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two type for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Fighter
newtype Knight = Knight Fighter
```